### PR TITLE
Bump reva deps

### PR DESCRIPTION
**Pull request by butonic**
bump all the deps!

and make tusd CORS headers configurable, see https://github.com/cs3org/reva/pull/4507

### CODE REVIEW

File: vendor/go-micro.dev/v4/api/options.go [hunk 1]
```diff
@@ -16,10 +19,19 @@ func NewOptions(opts ...Option) Options {
...
+// WithRouter sets the router to use e.g static or registry.
 func WithRouter(r router.Router) Option {
 	return func(o *Options) error {
 		o.Router = r
 		return nil
 	}
 }
+
+// WithRegistry sets the api's client and router to use registry.
+func WithRegistry(r registry.Registry) Option {
```
Comment (micbar) [inline vendor/go-micro.dev/v4/api/options.go:31]: @butonic Does that change break anything how we use the go micro client?

**Review by micbar — APPROVED** [review]
Did a review of the vendor folder.

Changes are looking good.

Please clarify that the go-micro client and Registry changes are already used in ocis.

If that is correct, ok from my side.

File: vendor/go-micro.dev/v4/api/options.go (hunk 1 above)
Comment (butonic) [reply vendor/go-micro.dev/v4/api/options.go:31]: WithRegistry is optional. We always set the registry in out ocis-pkg http and grpc client and server packages.

**Comment by butonic** [comment]
I see an error when stopping a large upload:
```json
{"level":"error","service":"frontend","pkg":"rhttp","traceid":"3c99e006acb683d3b43313b520b7b542","request-id":"37c1b3bc-a36f-4dfa-8d33-68eee87cb3c1","error":"Patch \"http://localhost:9158/data/tus/335e922d-c994-4eec-b4e3-65d988d2811f\": context canceled","time":"2024-02-19T13:34:00+01:00","message":"error doing PATCH request to data service"}
{"level":"error","service":"storage-users","pkg":"rhttp","traceid":"03e9501a2ea3a36c382d32549a4e704a","host":"127.0.0.1","method":"PATCH","uri":"/data/tus/335e922d-c994-4eec-b4e3-65d988d2811f","url":"/335e922d-c994-4eec-b4e3-65d988d2811f","proto":"HTTP/1.1","status":500,"size":15,"start":"19/Feb/2024:13:33:58 +0100","end":"19/Feb/2024:13:34:00 +0100","time_ns":1144378976,"time":"2024-02-19T13:34:00+01:00","message":"http"}
{"level":"error","service":"frontend","pkg":"rhttp","traceid":"3c99e006acb683d3b43313b520b7b542","host":"127.0.0.1","method":"PATCH","uri":"/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDMyMzgwLCJpYXQiOjE3MDgzNDU5ODAsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zMzVlOTIyZC1jOTk0LTRlZWMtYjRlMy02NWQ5ODhkMjgxMWYifQ.b6i-_kih05jLbuc_FIdfmORMJ76lignlOmoZvQTucdU","url":"/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDMyMzgwLCJpYXQiOjE3MDgzNDU5ODAsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zMzVlOTIyZC1jOTk0LTRlZWMtYjRlMy02NWQ5ODhkMjgxMWYifQ.b6i-_kih05jLbuc_FIdfmORMJ76lignlOmoZvQTucdU","proto":"HTTP/1.1","status":500,"size":0,"start":"19/Feb/2024:13:33:58 +0100","end":"19/Feb/2024:13:34:00 +0100","time_ns":1145671040,"time":"2024-02-19T13:34:00+01:00","message":"http"}
{"level":"error","service":"frontend","pkg":"rhttp","traceid":"8851badfc7074bd0921755e09113ca90","host":"127.0.0.1","method":"DELETE","uri":"/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDMyMzgwLCJpYXQiOjE3MDgzNDU5ODAsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zMzVlOTIyZC1jOTk0LTRlZWMtYjRlMy02NWQ5ODhkMjgxMWYifQ.b6i-_kih05jLbuc_FIdfmORMJ76lignlOmoZvQTucdU","url":"/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDMyMzgwLCJpYXQiOjE3MDgzNDU5ODAsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zMzVlOTIyZC1jOTk0LTRlZWMtYjRlMy02NWQ5ODhkMjgxMWYifQ.b6i-_kih05jLbuc_FIdfmORMJ76lignlOmoZvQTucdU","proto":"HTTP/1.1","status":501,"size":0,"start":"19/Feb/2024:13:34:01 +0100","end":"19/Feb/2024:13:34:01 +0100","time_ns":103835,"time":"2024-02-19T13:34:01+01:00","message":"http"}
```

**Comment by butonic** [comment]
the 501 on delete is fixed with https://github.com/cs3org/reva/pull/4527

**Comment by butonic** [comment]
It seems that was not the problem.

The failing web test:
```
1) Scenario: uploading a big file (when chunking is implemented this upload should be chunked) (attempt 1, retried) # features/webUIUpload/upload.feature:87
   ✔ Before # setup.js:38
   ✔ Before # setup.js:42
   ✔ Before # setup.js:53
   ✔ Before # setup.js:60
   ✔ Before # setup.js:71
   ✔ Before # stepDefinitions/filesContext.js:17
   ✔ Before # stepDefinitions/generalContext.js:149
   ✔ Before # stepDefinitions/generalContext.js:187
   ✔ Before # stepDefinitions/middlewareContext.js:46
   ✔ Given user "Alice" has been created with default attributes and without skeleton files in the server # stepDefinitions/middlewareContext.js:66
   ✔ And user "Alice" has created folder "simple-folder" in the server # stepDefinitions/middlewareContext.js:66
   ✔ And user "Alice" has uploaded file with content "initial content" to "lorem.txt" in the server # stepDefinitions/middlewareContext.js:66
   ✔ And user "Alice" has uploaded file with content "initial content" to "simple-folder/lorem.txt" in the server # stepDefinitions/middlewareContext.js:66
   ✔ And user "Alice" has logged in using the webUI # stepDefinitions/loginContext.js:68
   ✔ Given a file with the size of "30000000" bytes and the name "big-video.mp4" has been created locally in the middleware # stepDefinitions/middlewareContext.js:66
   ✔ When the user uploads a created file "big-video.mp4" using the webUI # stepDefinitions/filesContext.js:215
   ✔ Then no message should be displayed on the webUI # stepDefinitions/generalContext.js:70
   ✖ And file "big-video.mp4" should be listed on the webUI # stepDefinitions/filesContext.js:372
       AssertionError [ERR_ASSERTION]: An error occurred while running .getText() command on <//span[contains(@class, "oc-resource-name") and (@data-test-resource-name='big-video.mp4' or @data-test-resource-path='/big-video.mp4') and @data-test-resource-type='file']>: 
           at Proxy.checkFileName (/drone/src/webTestRunner/tests/acceptance/pageObjects/FilesPageElement/filesList.js:401:16)
           at processTicksAndRejections (node:internal/process/task_queues:95:5)
           at Proxy.waitForFileVisible (/drone/src/webTestRunner/tests/acceptance/pageObjects/FilesPageElement/filesList.js:379:7)
   - And as "Alice" the content of "big-video.mp4" in the server should be the same as the content of local file "big-video.mp4" # stepDefinitions/middlewareContext.js:66
   ✔ After # stepDefinitions/middlewareContext.js:60
   ✔ After # stepDefinitions/generalContext.js:205
   ✔ After # stepDefinitions/generalContext.js:174
   ✔ After # stepDefinitions/generalContext.js:115
   ✔ After # setup.js:104
   ✔ After # setup.js:100
   ✔ After # setup.js:96
   ✔ After # setup.js:89
   ✔ After # setup.js:80
```

But locally I cannot reporduce ... is the upload maybe too slow?

**Comment by ScharfViktor** [comment]
> But locally I cannot reporduce ... is the upload maybe too slow?

I could reproduce it: 
- uploading file less than 10 mb - works
- uploading file more than 10 mb fails

How reproduce: 
- switch to branch `bump-reva-deps`
- `make -C ocis clean generate build`
- start `ocis` binary
- upload file more than 10Mb

Actual: 

https://github.com/owncloud/ocis/assets/84779829/0a94b890-6dad-4003-9e76-59564de16218


ocis log: `OCIS_LOG_LEVEL=debug `

```
2024/02/19 17:07:39 http: TLS handshake error from [::1]:60182: remote error: tls: unknown certificate
2024-02-19T17:07:39+01:00 DBG rewrite hook found line=/Users/scharfviktor/Work/ocis/services/proxy/pkg/router/router.go:224 method=PATCH path=/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDQ1MjU5LCJpYXQiOjE3MDgzNTg4NTksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy9lOWRhYWIzYy03YWVjLTQxMzUtOTU5NC01NWUwNzNlMzMyZjEifQ.PPiw28uNt0UIljGDioesLwp-Ax-sWUekOaMbHPKPE4U policy=ocis prefix=/data routeType=prefix service=proxy
2024-02-19T17:07:39+01:00 INF skipping auth check for: /data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNzA4NDQ1MjU5LCJpYXQiOjE3MDgzNTg4NTksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy9lOWRhYWIzYy03YWVjLTQxMzUtOTU5NC01NWUwNzNlMzMyZjEifQ.PPiw28uNt0UIljGDioesLwp-Ax-sWUekOaMbHPKPE4U line=/Users/scharfviktor/Work/ocis/vendor/github.com/cs3org/reva/v2/internal/http/interceptors/auth/auth.go:195 pkg=rhttp service=frontend traceid=b4a4ce5b918bfd7909b811e4a6bf2fcf
```

**Comment by saw-jan** [comment]
And there is this error log
```json
{"level":"error","service":"frontend","pkg":"rhttp","traceid":"16995d691cd01911e970ca91f8fa40ee","request-id":"766dcb17-c21b-4f97-98ae-56b1a7ad572a","content-length":24493475,"transferred-bytes":0,"time":"2024-02-20T09:16:28+05:45","message":"content length vs transferred bytes mismatch"}
{"level":"error","service":"frontend","pkg":"rhttp","traceid":"6479db37bf424506b2915d322c6a6e39","request-id":"b6c4aa22-8378-442e-b12d-8696764e0dc6","content-length":24493475,"transferred-bytes":0,"time":"2024-02-20T09:16:28+05:45","message":"content length vs transferred bytes mismatch"}
```

**Comment by butonic** [comment]
@saw-jan hm those lines are now logged by the datagateway in the frontent service when the content length does not match the actual body length:
```

	if httpRes.Header.Get("Content-Length") != "" {
		i, err := strconv.ParseInt(httpRes.Header.Get("Content-Length"), 10, 64)
		if err != nil {
			log.Error().Err(err).Str("content-length", httpRes.Header.Get("Content-Length")).Msg("invalid content length in dataprovider response")
		}
		if i != c {
			log.Error().Int64("content-length", i).Int64("transferred-bytes", c).Msg("content length vs transferred bytes mismatch")
		}
	}
```
But good hint!

I thought maybe I missed the `make -C ocis clean generate build` step ... but I don't get a 403 Forbidden. I need to check if I set an env var that affects this, because I have a differnt domain configured, which also sets CORS headers ...

👀

**Comment by butonic** [comment]
I accidentially checked in my domain as the CORS allowed origin for tus uploads. Changed it to `*` as all the other CORS allow origin defaults.

**Comment by butonic** [comment]
I assume these failures are caused by disableing resharing in the sharing manager:
```

  Scenario: sharing parent folder to user with all permissions and its child folder to group with read permission then check reshare operation # /drone/src/tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareReceivedInMultipleWays.feature:265

    Given group "grp1" has been created                                                                                                        # FeatureContext::groupHasBeenCreated()

    And user "Carol" has been created with default attributes and without skeleton files                                                       # FeatureContext::userHasBeenCreatedWithDefaultAttributes()

    And user "Carol" has created the following folders                                                                                         # FeatureContext::userHasCreatedFollowingFolders()

      | path                  |

      | /parent               |

      | /parent/child1        |

      | /parent/child1/child2 |

    And user "Alice" has been added to group "grp1"                                                                                            # FeatureContext::userHasBeenAddedToGroup()

    And user "Brian" has been added to group "grp1"                                                                                            # FeatureContext::userHasBeenAddedToGroup()

    And user "Carol" has created a share with settings                                                                                         # FeatureContext::userHasCreatedAShareWithSettings()

      | path        | /parent |

      | shareType   | user    |

      | shareWith   | Brian   |

      | permissions | all     |

    And user "Carol" has created a share with settings                                                                                         # FeatureContext::userHasCreatedAShareWithSettings()

      | path        | /parent/child1 |

      | shareType   | group          |

      | shareWith   | grp1           |

      | permissions | read           |

    When user "Brian" creates a share using the sharing API with settings                                                                      # FeatureContext::userCreatesAShareWithSettings()

      | path        | /Shares/parent |

      | shareType   | user           |

      | shareWith   | Alice          |

      | permissions | read           |

    Then the HTTP status code should be "200"                                                                                                  # FeatureContext::thenTheHTTPStatusCodeShouldBe()

    And the OCS status code should be "100"                                                                                                    # OCSContext::theOCSStatusCodeShouldBe()

      OCS status code is not any of the expected values 100 got 104

      Failed asserting that an array contains '104'.

    And as "Brian" folder "/Shares/child1" should exist                                                                                        # FeatureContext::asFileOrFolderShouldExist()

    And as "Alice" folder "/Shares/child1" should exist                                                                                        # FeatureContext::asFileOrFolderShouldExist()

    And as "Alice" folder "/Shares/parent" should exist                                                                                        # FeatureContext::asFileOrFolderShouldExist()
```

So I changed it to enabled again.

AFAICT the testsuite uses `all` in the permissions ... which is different when using disabling resharing, because then the sharing permission (16) cannot be granted.

**Comment by butonic** [comment]
looking into this:
```gherkin
 @issue-1233
  Scenario Outline: get a share with a user that didn't receive the share                # /drone/src/tests/acceptance/features/coreApiShareOperationsToShares1/gettingShares.feature:134
    Given using OCS API version "<ocs_api_version>"                                      # FeatureContext::usingOcsApiVersion()
    And user "Carol" has been created with default attributes and without skeleton files # FeatureContext::userHasBeenCreatedWithDefaultAttributes()
    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"      # FeatureContext::userHasUploadedAFileWithContentTo()
    And user "Alice" has shared file "textfile0.txt" with user "Brian"                   # FeatureContext::userHasSharedFileWithUserUsingTheSharingApi()
    When user "Carol" gets the info of the last share using the sharing API              # FeatureContext::userGetsInfoOfLastShareUsingTheSharingApi()
    Then the OCS status code should be "404"                                             # OCSContext::theOCSStatusCodeShouldBe()
    And the HTTP status code should be "<http_status_code>"                              # FeatureContext::thenTheHTTPStatusCodeShouldBe()

    Examples:
      | ocs_api_version | http_status_code |
      | 1               | 200              |
        Failed step: Then the OCS status code should be "404"
        OCS status code is not any of the expected values 404 got 998
        Failed asserting that an array contains '998'.
      | 2               | 404              |
        Failed step: Then the OCS status code should be "404"
        OCS status code is not any of the expected values 404 got 998
        Failed asserting that an array contains '998'.
```

**Comment by ScharfViktor** [comment]
> looking into this:

this test is in expected failures file, do you want to fix it too? 

I see that another tests is reason the failed CI. (where we get `104` instead of `100`)
If you need helping with test debug, just ping me

**Comment by saw-jan** [comment]
Yeah, the failure is during creating share/reshare
```feature
    And user "Carol" has been added to group "grp1" 
    And user "Alice" has shared folder "common" with group "grp1" 
    And user "Alice" has shared file "textfile0.txt" with user "Carol"
      OCS status code is not any of the expected values 100,200 got 104
      Failed asserting that an array contains '104'.
```
And AFAIK, `104` means something doesn't exit

**Comment by butonic** [comment]
🤦‍♂️ damn looking at the wrong test ...

**Comment by butonic** [comment]
looking at 
```gherkin
  @issue-2441
  Scenario Outline: shares shared to deleted group should not be available                            # /drone/src/tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:440
    Given using OCS API version "<ocs_api_version>"                                                   # FeatureContext::usingOcsApiVersion()
    And these users have been created with default attributes and without skeleton files:             # FeatureContext::theseUsersHaveBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
      | username |
      | Brian    |
      | Carol    |
    And group "grp1" has been created                                                                 # FeatureContext::groupHasBeenCreated()
    And user "Brian" has been added to group "grp1"                                                   # FeatureContext::userHasBeenAddedToGroup()
    And user "Carol" has been added to group "grp1"                                                   # FeatureContext::userHasBeenAddedToGroup()
    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"   # FeatureContext::userHasUploadedAFileWithContentTo()
    And user "Alice" has shared file "/textfile0.txt" with group "grp1"                               # FeatureContext::userHasSharedFileWithGroupUsingTheSharingApi()
    When user "Alice" sends HTTP method "GET" to OCS API endpoint "/apps/files_sharing/api/v1/shares" # OCSContext::userSendsToOcsApiEndpoint()
    Then the OCS status code should be "<ocs_status_code>"                                            # OCSContext::theOCSStatusCodeShouldBe()
    And the HTTP status code should be "200"                                                          # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And the fields of the last response to user "Alice" sharing with group "grp1" should include      # FeatureContext::checkFieldsOfLastResponseToUser()
      | share_with  | grp1                  |
      | file_target | /Shares/textfile0.txt |
      | path        | /textfile0.txt        |
      | uid_owner   | %username%            |
    Then as "Brian" file "/Shares/textfile0.txt" should exist                                         # FeatureContext::asFileOrFolderShouldExist()
    And as "Carol" file "/Shares/textfile0.txt" should exist                                          # FeatureContext::asFileOrFolderShouldExist()
    When the administrator deletes group "grp1" using the Graph API                                   # GraphContext::userDeletesGroupUsingTheGraphApi()
    And user "Alice" sends HTTP method "GET" to OCS API endpoint "/apps/files_sharing/api/v1/shares"  # OCSContext::userSendsToOcsApiEndpoint()
    Then the OCS status code should be "<ocs_status_code>"                                            # OCSContext::theOCSStatusCodeShouldBe()
    And the HTTP status code should be "200"                                                          # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And file "/textfile0.txt" should not be included as path in the response                          # FeatureContext::checkSharedFileAsPathNotInResponse()
    And as "Brian" file "/Shares/textfile0.txt" should not exist                                      # FeatureContext::asFileOrFolderShouldNotExist()
    And as "Carol" file "/Shares/textfile0.txt" should not exist                                      # FeatureContext::asFileOrFolderShouldNotExist()

    Examples:
      | ocs_api_version | ocs_status_code |
      | 1               | 100             |
        │ path is unexpectedly set with value '/textfile0.txt'
        │ 
        Failed step: And file "/textfile0.txt" should not be included as path in the response
        'path' value '/textfile0.txt' was unexpectedly found in response
        Failed asserting that true is false.
      │
      ╳  HTTP status code 404 is not the expected value 204
      ╳  Failed asserting that 404 matches expected 204.
      │
      └─ @AfterScenario # FeatureContext::afterScenario()
      | 2               | 200             |
        │ path is unexpectedly set with value '/textfile0.txt'
        │ 
        Failed step: And file "/textfile0.txt" should not be included as path in the response
        'path' value '/textfile0.txt' was unexpectedly found in response
        Failed asserting that true is false.
      │
      ╳  HTTP status code 404 is not the expected value 204
      ╳  Failed asserting that 404 matches expected 204.
      │
      └─ @AfterScenario # FeatureContext::afterScenario()
```

**Comment by butonic** [comment]
@saw-jan hm 104 means forbidden. At least that is what it should mean:
```
// MetaForbidden is an error response with code 104
var MetaForbidden = Meta{Status: "", StatusCode: 104, Message: "Forbidden"}
```

But I ran into https://github.com/owncloud/ocis/issues/1233 which expects an OCS 404 status for a non existing share, which IMNSHO should be an OCS 998 status ... but OCS is lagacy, so ... whatever. Aligned to be as broken as oc10 with https://github.com/cs3org/reva/pull/4529

Digging further into the wrong Create response...

**Comment by butonic** [comment]
There is a bug when creating shares ... we still add the share permission somewhere ... 👀

**Comment by butonic** [comment]
The error we get on the OCS API is 

````go
// MessageShareExists is used when a user tries to create a new share for the same user
var MessageShareExists = "A share for the recipient already exists"
``

which is only used when the Share() call returns an ALREADY_EXISTS code:
```go
	createShareResponse, err := client.CreateShare(ctx, req)
	if err != nil {
		return nil, &ocsError{
			Code:    response.MetaServerError.StatusCode,
			Message: "error sending a grpc create share request",
			Error:   err,
		}
	}
	if createShareResponse.Status.Code != rpc.Code_CODE_OK {
		logger.Debug().Interface("Code", createShareResponse.Status.Code).Str("message", createShareResponse.Status.Message).Msg("grpc create share request failed")
		switch createShareResponse.Status.Code {
		case rpc.Code_CODE_NOT_FOUND:
			return nil, &ocsError{
				Code:    response.MetaNotFound.StatusCode,
				Message: "not found",
				Error:   nil,
			}
		case rpc.Code_CODE_ALREADY_EXISTS:
			return nil, &ocsError{
				Code:    response.MetaForbidden.StatusCode,
				Message: response.MessageShareExists,
				Error:   nil,
			}
 // ...
```

But why would it already exist ...

**Comment by butonic** [comment]
The jsoncs3 manager does return ALREADY_EXISTS if it can read the share, based on resoucreid and grantee:
```

	// check if share already exists.
	key := &collaboration.ShareKey{
		// Owner:      md.Owner, owner no longer matters as it belongs to the space
		ResourceId: md.Id,
		Grantee:    g.Grantee,
	}

	_, err := m.getByKey(ctx, key)
	if err == nil {
		// share already exists
		err := errtypes.AlreadyExists(key.String())
		span.RecordError(err)
		span.SetStatus(codes.Error, err.Error())
		return nil, err
	}
```
But that test does not even share the same resource twice...

**Comment by butonic** [comment]
Ran into https://github.com/cs3org/reva/pull/4530.

It should not solve the test failure ... but why does that `granteeExists` check return `false` ... when the Share request returns an ALREADY_EXISTS error?!?

**Comment by ScharfViktor** [comment]
`Failed asserting that an array contains '104'` only reproducible using api test. E2e tests are green.

error occurs only AFTER we perform step sharing to group

try to run this test: 
```
 Scenario: 
    Given using spaces DAV path
    And group "grp1" has been created
    And user "Brian" has been added to group "grp1"
    And user "Alice" has uploaded a file inside space "Alice Hansen" with content "some content" to "textfile.txt"
    And user "Alice" has uploaded a file inside space "Alice Hansen" with content "some content2" to "textfile1.txt"
    And user "Alice" has uploaded a file inside space "Alice Hansen" with content "some content3" to "textfile2.txt"
    And user "Alice" creates a share inside of space "Alice Hansen" with settings:
      | path      | textfile.txt |
      | shareWith | grp1        |
      | shareType  | 1                        |
      | role      | editor       |
    Then the HTTP status code should be "200"
    And the OCS status code should be "200"

    And user "Alice" has shared file "textfile2.txt" with user "Brian"
    Then the HTTP status code should be "200"
    And the OCS status code should be "200"
   
   And user "Alice" creates a share inside of space "Alice Hansen" with settings:
      | path      | textfile1.txt |
      | shareWith | Brian        |
      | role      | editor       |
    Then the HTTP status code should be "200"
    And the OCS status code should be "200"

```

**Comment by butonic** [comment]
hm

```
{"level":"error","service":"sharing","pkg":"rgrpc","traceid":"3f3f6f6c205e1e716fb799090b2bf1b3","hostname":"965c0b73b289","userID":"eeaedb89-922c-411f-a0fe-3751acd023d4","spaceID":"878b1b24-8445-46d7-b1e3-001899415023:59284822-8b4e-464b-8605-48c416260344","error":"error: already exists: already exists","time":"2024-02-20T11:12:04Z","message":"persisting added received share failed"}
```

**Comment by butonic** [comment]
Added some [more ocs logging](https://github.com/cs3org/reva/pull/4531/files):
```json
{"level":"error","service":"sharing","pkg":"rgrpc","traceid":"7db3c8ade71b0c1ea51c71c572c11413","hostname":"c4a1b5c5e858","userID":"db8541d4-366e-44ae-812e-26d7cb4c9def","spaceID":"9812b81b-b85e-4549-8fe3-606dbfb53a4c:8c0b71b7-2e7b-4c2a-a8bb-76fdd22c0e7a","error":"error: already exists: already exists","time":"2024-02-20T19:38:29Z","message":"persisting added received share failed"}

{"level":"error","service":"frontend","pkg":"rhttp","traceid":"4f089c03c0c4b030ad7b15777bef5bba","request-id":"coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:474-486","error":"error creating share: error: already exists: already exists","ocs_msg":"A share for the recipient already exists","time":"2024-02-20T19:38:29Z","message":"writing ocs error response"}

{"level":"error","service":"frontend","pkg":"rhttp","traceid":"0c5ac2f7ed7df3073acbb811f0e2526a","request-id":"coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:500-509","error":"cannot set the requested share permissions","ocs_msg":"Cannot set the requested share permissions","time":"2024-02-20T19:38:33Z","message":"writing ocs error response"}

{"level":"error","service":"frontend","pkg":"rhttp","traceid":"96da2e73273862b7803ff55e01536e33","request-id":"coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:516-525","error":"cannot set the requested share permissions","ocs_msg":"Cannot set the requested share permissions","time":"2024-02-20T19:38:37Z","message":"writing ocs error response"}

{"level":"error","service":"frontend","pkg":"rhttp","traceid":"6f45ae8e91300017fd2d97c4e6d91687","request-id":"coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:532-543","error":"cannot set the requested share permissions","ocs_msg":"Cannot set the requested share permissions","time":"2024-02-20T19:38:42Z","message":"writing ocs error response"}
```

**Comment by butonic** [comment]
Hah, got it. This is caused by https://github.com/cs3org/reva/pull/4528 which now actually checks the IfNoneMatch header.

And `persist()` in the jsoncs3 share manager also uses it:
```go

	ur := metadata.UploadRequest{
		Path:        jsonPath,
		Content:     createdBytes,
		IfMatchEtag: rss.etag,
	}
	// when there is no etag in memory make sure the file has not been created on the server, see https://www.rfc-editor.org/rfc/rfc9110#field.if-match
	// > If the field value is "*", the condition is false if the origin server has a current representation for the target resource.
	if rss.etag == "" {
		ur.IfNoneMatch = []string{"*"}
	}

	_, err = c.storage.Upload(ctx, ur)
```

At least I think that causes this. @ScharfViktor looking at the logic the code tries to write the `userJSONPath(userID)` which might not exist if the user has just been created! HAH!

**Comment by ScharfViktor** [comment]
> At least I think that causes this. @ScharfViktor looking at the logic the code tries to write the `userJSONPath(userID)` which might not exist if the user has just been created! HAH!

yes, I can confirm it using UI. 

- create new user 
- share file or folder to new user

Actual: 
Screenshot 2024-02-20 at 21 21 49


But then I can't understand why the e2e tests and all the api tests don't fail. After each test we delete user `Alice` and create new user `Alice` (different uuid). That should be enough to get an error.

**Comment by butonic** [comment]
It also has to do with the group shares. I'll take a look tomorrow.

**Pull request merged**